package.json

```diff
@@ -9,7 +9,8 @@
     "path-posix": "^1.0.0",
     "phosphor-disposable": "^1.0.5",
     "phosphor-signaling": "^1.2.0",
-    "requirejs": "^2.2.0"
+    "requirejs": "^2.2.0",
+    "url": "^0.11.0"
```

Comment:
FYI, I noticed that both url and the path module we are using are missing upstream bugfixes from npm (cf. https://github.com/defunctzombie/node-url/issues/24). They may be better than if we tried to write something from scratch, though...